### ISSUE TITLE
Error when updating WYSIWYG interface after upgrade to v8.5.0

### ISSUE THREAD

**Issue by blob100** [directus/api#1722]
I just updated via GIT from v8.4.0 to v8.5.0
On the app I go to `Settings` > `Collections & Fields` and select a current collection.
I click on an existing WYSIWYG interface and add fullscreen from the options.
On the front end I get an error:
> Trouble processing request. Try again after refreshing the page

On the console I get this error:
`Error: "Statement could not be executed (42000 - 1064 - You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near '(16777215) Null' at line 1)"
    a API.js:39
    request API.js:253
 error.js:6:22
    Ud error.js:6
    VueJS 2
    e fields.vue:448
    l runtime.js:45
    O runtime.js:271
    a runtime.js:97
    n asyncToGenerator.js:3
    m asyncToGenerator.js:29`

Changing `MySQL Datatype` from `MEDIUMTEXT` to `TEXT` at the same time as adding fullscreen option allows the updated setting to be saved.  

When I click back into the interface settings the `MySQL Datatype` reverts back to `MEDIUMTEXT`, but the fullscreen option is preserved offering a workaround.

**Comment by nachogarcia** [comment]
I finally updated (I'm using docker) from 8.3 to 8.5.1 and had no issues. (mysql 5.7)

**Comment by dur41d** [comment]
I was able to reproduce it in my [ddev](https://docs.directus.io/advanced/ddev.html) environement. The error happens in `Directus\Api\Routes\Fields->update`.  For some reason it tries to update the table schema which i don't think is correct. This statement is failing:

```ALTER TABLE `news`  MODIFY COLUMN `ewwe` LONGTEXT(4294967295) Null```

Any idea why it's trying the change the table schema in the first place?

Server: db via TCP/IP
Server type: MariaDB
Server version: 10.2.27-MariaDB-1:10.2.27+maria~bionic-log - mariadb.org binary distribution
Protocol version: 10
User: db@172.19.0.4
Server charset: UTF-8 Unicode (utf8)

**Comment by dur41d** [comment]
BTW, I saw this statement in `Directus\Api\Routes\Fields` and it don't think it's a practice to catch and throw and SQLException. It does nothing but losing the stack trace of the original exception. I think the try block should be removed.

```php
 try {
            $responseData = $service->changeColumn(
                $request->getAttribute('collection'),
                $request->getAttribute('field'),
                $request->getParsedBody(),
                $request->getQueryParams()
            );
        } catch (\Exception $e) {
            throw new SQLException($e->getMessage());
        }
```

**Comment by dur41d** [comment]
The statement would work the length is removed like this:

```ALTER TABLE `news` MODIFY COLUMN `ewwe` LONGTEXT Null```

I think the problem is with putting the length in the LONGTEXT type field. I think the length should be omitted from this type of field.

**Comment by WoLfulus** [comment]
@dur41d are you able to get a stacktrace? it might help tracing the issue

thanks :)

**Comment by dur41d** [comment]
I created a PR for this.

https://github.com/directus/api/pull/1727

**Comment by blob100** [comment]
Sorry for the delay - timezones.
The original error was when using MySQL Version: 10.1.40-MariaDB-cll-lve

**Comment by benhaynes** [comment]
Not sure if this is the cause, but Directus says in the docs that MariaDB 10.2+ is required.

**Comment by dur41d** [comment]
I updated to the MariaDB 10.4 and the error still happens.

**Comment by dur41d** [comment]
According to [this](https://dev.mysql.com/doc/refman/8.0/en/string-type-syntax.html) only TEXT allows optional length parameter. TINYTEXT, MEDIUMTEXT, LONGTEXT don't allow length variable. I'll update the PR to set length=null for those data types.

**Comment by dur41d** [comment]
I was able to reproduce this with MySQL v8.0.17. The reason is that directus is producing an invalid alter statement for `longtext` and `mediumtext` datatypes. This is not new bug and I was able to reproduce it with v8.4. I think it's just that nobody tried to update a wysiwyg field before adding the fullscreen option. The erroneous statement looks like this:

```ALTER TABLE `news` MODIFY COLUMN `[column name]` LONGTEXT(4294967295) Null```

This is the correct statement without the length:


```ALTER TABLE `news` MODIFY COLUMN `[column name]` LONGTEXT Null```

steps to reproduce:

1. Create a collection with wysiwyg field
2. Try to update the field by **unchecking** one of the toolbar options.

I'm not sure why directus is issuing an alter statement to begin with. I traced the problem to this function in `TableService.php` that is evaluating `true` but it does not make sense to me as the column schema does not need an update when changing the toolbar options. But i'm still new to the code base and this is beyond me to fix for now.

```php
protected function shouldUpdateSchema(array $data)
    {
        // NOTE: If any of these attributes exists the database needs to update the column
        return ArrayUtils::containsSome($data, [
            'type',
            'datatype',
            'unique',
            'primary_key',
            'auto_increment',
            'length',
            'note',
            'signed',
            'nullable',
        ]);
    }
```
